torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -472,7 +472,10 @@ def __init__(
                 "ignored_states": self._ignored_params,
                 "device_mesh": device_mesh,
             }
-            if sharding_strategy in HYBRID_SHARDING_STRATEGIES:
+            if (
+                sharding_strategy in HYBRID_SHARDING_STRATEGIES
+                and device_mesh is not None
```

Comment:
This might just be a typo? I think we want `device_mesh is None` (i.e. the user did not pass `device_mesh)` -- then we forward the `process_group` constructed by the root to the children.

I opened a PR with this here with some basic test: https://github.com/pytorch/pytorch/pull/119064